### DISCUSSION TITLE
BUG: Postgres database slows down for admin section after adding large amount of users

### DISCUSSION THREAD

**Discussion by beastech** [open-webui/open-webui#7013] [Ideas]
## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.35

- **Operating System:** Azure App Service

-**Database:** External Postgres database

**Confirmation:**

- [*] I have read and followed all the instructions provided in the README.md.
- [*] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [*] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When using the built in sqlite database click on "Admin Panel", users appear quickly on the screen.  Changing between pending, user, and admin is pretty much instantaneous, even with 100+ users.

## Actual Behavior:

When using an external postgres database with 100+ users the system begins to bog down when clicking on "Admin Panel" and takes a long time for it to load the table on the screen.  Making a change to a user, such as switching roles takes just as long.  The more users added, the longer it takes to load.

## Description

**Bug Summary:**
External postgres database takes a very long time to load users in the admin section if you have a lot of users.  This may happen with external sqlite database as well but I do not have this option available to me to test in my current environment.

## Reproduction Details
external postgres database. 
environment variable: DATABASE_URL=postgres://<server>:<port>/<database>
I'm using Azure and hosting as an app service, pointing to a postgres database in the same subscription.  The only option I have on the database is sslmode=require and user\pass.

**Steps to Reproduce:**
Set up external postgres database.
Connect OWUI to database using environment variable: DATABASE_URL=postgres://<server>:<port>/<database>
Add 100 users under "Admin Panel".
Click out of admin panel, go back to admin panel and wait.  Eventually the users will appear.

**Additional Info***
The database is working, the admin panel is the only part that seems to have an issue.  When users are prompting, getting responses, logging in, etc. everything seems fine.

**Comment by tjbck** [comment]
Optimisation PR welcome!

**Reply by GanizaniSitara** [reply]
seeing the same on version 0.5.4, with 5 users

**Reply by harrywang** [reply]
yes. same issue here with 0.5.18, external postgres db slows down significantly - don't know why.

**Reply by harrywang** [reply]
I hope someone run open webui with postgres can share the optimizations they did. I just changed the connection pool size to 30 and the speed get much better but still feels much slower compared with sqlite.

**Comment by crelocks** [comment]
Is this still an issue? I am wondering to migrate my db to postgres. But I see these issues and not sure if I should make that move?

**Reply by abiari** [reply]
it still is, I'm on v0.5.16, we went to a 1000 users now and the app as a whole feels slower with the users dashboard taking 20s to load.

**Reply by crelocks** [reply]
@tjbck Can you please help us with some recommendations on how to solve this? Is this some sort of an indexing issue?

**Comment by dekiesel** [comment]
Would you mind sharing a bit more detail about how you connected openwebui to postgres?
I am running owui in an azure container app and I created a postgres instance in azure. I then connected them via a "service connector", but I am struggling to understand how to structure `DATABASE_URL` now.
Any help is appreciated!
